sui_types/src/crypto.rs

```diff
@@ -70,6 +71,27 @@ impl<'de> Deserialize<'de> for KeyPair {
     }
 }
 
+impl FromStr for KeyPair {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let value = Base64::decode_vec(s).map_err(|e| anyhow!("{}", e.to_string()))?;
+        let key = dalek::Keypair::from_bytes(&value).map_err(|e| anyhow!("{}", e.to_string()))?;
+        Ok(KeyPair {
+            key_pair: key,
+            public_key_cell: OnceCell::new(),
+        })
+    }
+}
+
+impl std::fmt::Debug for KeyPair {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::result::Result<(), std::fmt::Error> {
+        let s = Base64::encode_string(&self.key_pair.to_bytes());
+        write!(f, "{}", s)?;
+        Ok(())
+    }
+}
+
```

Comment:
This prints a private key. Please make sure you only print the public key and elide the private one.

Reply:
If you need to print, e.g. a private key you just generated, you can add a way of printing it that only triggers in the control flow following this automatic generation (e.g. by inlining the implementation you have here). This adds a `Debug` instance on all private keys, essentially, and we can't have it over the whole code base.